src/integrationtest/java/infrastructure/SeleniumExtensions.java

```diff
@@ -81,8 +88,33 @@ public static void performADLogin(WebDriver driver, User user){
         waitForElementToBeVisibleAndEnable(driver, by).sendKeys(user.getPassword());
 
         LOG.info("Loggin in ... click submit");
-         waitForElementToBeVisibleAndEnable(driver, new By.ById(fields.getPasswordSigInButtonId())).
+        waitForElementToBeVisibleAndEnable(driver, new By.ById(fields.getPasswordSigInButtonId())).
                 click();
+
+        try {
+            checkAuthenticationCompletePage(driver);
+            return;
+        } catch (TimeoutException ex) {
```

Comment:
If there is a timeout exception, should we erroring out with a helpful message? There is nothing in the catch block here or in line 105

Reply:
depending on types of account and authority authentication can be completed or question about "Staying logged in" might appears. Both option are valid.